src/sentry/web/frontend/accounts.py

```diff
@@ -123,7 +123,13 @@ def recover_confirm(request, user_id, hash):
 @login_required
 def start_confirm_email(request):
     has_unverified_emails = request.user.has_unverified_emails()
-    if has_unverified_emails:
+
+    if 'email' in request.GET:
```

Comment:
one other thing, we should consider making this POST, because otherwise someone can easily embed this URL on another site and repeatedly cause emails to be generated